Addition of USERNAME and PASSWORD to environment variables to create Owner user.

bikash119 opened the pull request:
Included default username/password to environment variable to create owner user.

# Description
<!-- Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. -->
- Added USERNAME and PASSWORD to the environment variables. This change will allow the user to login to Argilla UI with owner user credentials. 

Closes #5434

**Type of change**
<!--  Please delete options that are not relevant. Remember to title the PR according to the type of change  -->

- Bug fix (non-breaking change which fixes an issue)


**How Has This Been Tested**
<!--  Please add some reference about how your feature has been tested.  -->
## In dev machine
- install minikube ( local k8s cluster)
- Create k8s resources ( kubect apply -f <resource.yaml> )
- Configure port forwarding ( kubectl port-forward service/argilla-server 6900:6900 )
- Open Argilla UI ( http://localhost:6900)
- Login with username: argilla , password : 12345678
- The user will get authenticated and will be redirected to http://localhost:6900/datasets screen.

**Checklist**
<!--  Please go over the list and make sure you've taken everything into account -->

- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

bikash119 committed:
Included default username/password to environment variable to create owner user.

bikash119 committed:
Added the details of the fix for issue https://github.com/argilla-io/argilla/issues/5434

davidberenstein1957 commented on argilla/CHANGELOG.md:
```diff
@@ -18,6 +18,10 @@ These are the section headers that we use:
+### Fixed
+
+- Fixed the deployment yaml used to create a new Argilla server in K8s. Added `USERNAME` and `PASSWORD` to the environment variables of pod template. ([#5434](https://github.com/argilla-io/argilla/issues/5434))
+
```
```suggestion
```

davidberenstein1957 reviewed (commented):
Hi, this looks good. Can you also confirm this has been added to the right documentation in the argilla-server directory and potentially also argilla docs itself. https://docs.argilla.io/latest/reference/argilla-server/configuration/
and the readme for the server. Thanks for the contribution :)

bikash119 committed:
Moved entry to Unreleased section

bikash119 committed:
Added USERNAME and PASSWORD to Authentication section of Server configuration

bikash119 committed:
Merge branch 'develop' into bug/5434

davidberenstein1957 commented on examples/deployments/k8s/argilla-server-deployment.yaml:
```diff
@@ -29,6 +29,10 @@ spec:
+            - name: USERNAME
+              value: "argilla"
+            - name: PASSWORD
+              value: "12345678"
```
```suggestion
            - name: USERNAME
              value: "argilla"
            - name: PASSWORD
              value: "12345678"
```
I think we want to comment these out by default.

bikash119 replied:
@davidberenstein1957 : Thank you for the remark. If I understand correctly, you want these environment variables to be commented like 
```
 #           - name: USERNAME
 #            value: "argilla"
 #           - name: PASSWORD
 #             value: "12345678"
```
or I should add comments to these environment variables?

If these env variables are commented out, then the default owner user will not get created.

davidberenstein1957 replied:
Yes, I think people need to be made aware of the action, because enforcing this will otherwise lead to security risks.

bikash119 replied:
Commented the default USERNAME and PASSWORD for owner user
[Commented the USERNAME and PASSWORD field](https://github.com/argilla-io/argilla/pull/5506/commits/811fb3df1f58ff90a06ec0d243f2e91eec9204db#diff-75693acb7abb4409fa1d7c8fe68045e41a1731d228bac6cf0f154a2f5c94401eL32)

davidberenstein1957 reviewed (commented):
@bikash119, one more remark. After that, I will merge the PR :)

davidberenstein1957 commented on argilla/docs/reference/argilla-server/configuration.md:
```diff
@@ -51,6 +51,10 @@ You can set the following environment variables to further configure your server
 - `ARGILLA_AUTH_SECRET_KEY`: The secret key used to sign the API token data. You can use `openssl rand -hex 32` to generate a 32 character string to use with this environment variable. By default a random value is generated, so if you are using more than one server worker (or more than one Argilla server) you will need to set the same value for all of them.
+- `USERNAME`: If provided, the owner username (Default: `""`).
+- `PASSWORD`: If provided, the owner password.
+If `USERNAME` and `PASSWORD` are provided, the owner user will be created with these credentials on the server startup (Default: `""`).
```
```suggestion
- `USERNAME`: If provided, the owner username (Default: `None`).
- `PASSWORD`: If provided, the owner password.
If `USERNAME` and `PASSWORD` are provided, the owner user will be created with these credentials on the server startup (Default: `None`).
```

bikash119 committed:
Update argilla/docs/reference/argilla-server/configuration.md

Co-authored-by: David Berenstein <david.m.berenstein@gmail.com>

bikash119 committed:
Update argilla-server-deployment.yaml

Commented the USERNAME and PASSWORD field. This will enforce users to set USERNAME and PASSWORD for Owner explicitly. 
Thank you @david for the review comments.

bikash119 committed:
Merge branch 'develop' into bug/5434

davidberenstein1957 committed:
Update configuration.md

davidberenstein1957 committed:
Update argilla-server-deployment.yaml
